modules/contact/src/actions/ContactAction.C

```diff
@@ -368,72 +395,161 @@ ContactAction::addMortarContact()
       _problem->addVariable(var_type, variable_name, params);
     };
 
-    // Normal contact: same family/order as primal.
+    // To avoid instabilities in the solution and obtain the full benefits of a variational
+    // enforcement, use of dual mortar is strongly recommended.
+
+    // Normal contact:
+    //   Lagrange family with order one less than primal, but by necessity with a min
+    //   order of 1 (we don't have zeroth order Lagrange). We must use a Lagrange basis because we
+    //   need dofs at nodes in order to enforce the zero-penetration constraint
+    //
+    if (_mortar_approach == MortarApproach::Weighted)
+      addLagrangeMultiplier(
+          normal_lagrange_multiplier_name, 1, false, getParam<Real>("normal_lm_scaling"));
+    else if (_mortar_approach == MortarApproach::Legacy)
+      addLagrangeMultiplier(
+          normal_lagrange_multiplier_name, 1, false, getParam<Real>("normal_lm_scaling"));
```

Comment:
this seems to be doing the same thing in both branches

```suggestion
    if (_mortar_approach == MortarApproach::Weighted || 
        _mortar_approach == MortarApproach::Legacy)
      addLagrangeMultiplier(
          normal_lagrange_multiplier_name, 1, false, getParam<Real>("normal_lm_scaling"));
```

Reply:
Probably we can just fully remove the `if` statement here since it is enclosed in the `mortar` formulation.